Uncompilable Java code generated for nested !type

### Expected behaviour
`reljam model` should create compilable code for nested types.

### Actual behaviour
Generated code does not compile

### Steps to reproduce the behaviour
```
> cat article.sysl
ArticleModel [version="0.0.1", package="io.sysl.bug"]: 
    !type Article: 
        content <: Content

    !type Content:
        Text <: string
> reljam --out gen model /article ArticleModel
> javac gen/io/sysl/bug/*.java
gen/io/sysl/issue66/ArticleModelJsonDeserializer.java:86: error: cannot find symbol
                case "content": entity.setContent(p.getCurrentToken() == JsonToken.VALUE_NULL ? null : deserialize(p, entity.getcontent()));  break;
                                                                                                                            ^
  symbol:   method getcontent()
  location: variable entity of type Article
1 error
```
[ internal: SYSL-35 ]

--- a/src/sysl/exporters/json_out/serializer.py
+++ b/src/sysl/exporters/json_out/serializer.py
@@ -396,7 +396,7 @@ def deserializer(context):
                                             f.set.type_ref.ref.path[-1])
                                     elif f.WhichOneof('type') == 'type_ref':
                                         extra = 'deserialize(p, entity.get{}())'.format(
-                                            jfname)
+                                            java.CamelCase(jfname))
                                     jsontype = ''
                                 else:
                                     raise RuntimeError(

--- a/test/java/article.sysl
+++ b/test/java/article.sysl
@@ -1,0 +1,10 @@
+# This Sysl file generates Java output to test "bad Java Case" issue
+# see: https://github.com/anz-bank/sysl/issues/66
+# <repo>/test/java > reljam --out gen model /article Article
+ArticleModel [package="io.sysl.issue66"]:
+
+    !type Article:
+        content <: Content
+
+    !type Content:
+        Text <: string

--- a/test/java/build.gradle
+++ b/test/java/build.gradle
@@ -24,7 +24,12 @@ task generateSysl(type:Exec) {
   }
 }
 
+task generateArticle(type:Exec) {
+   commandLine "reljam",  "--out", "$tempDirectory", "model", "/article", "ArticleModel"
+}
+
 compileJava.dependsOn generateSysl
+compileJava.dependsOn generateArticle
 
 sourceSets {
     main {

--- a/test/java/src/test/java/TestArticle.java
+++ b/test/java/src/test/java/TestArticle.java
@@ -18,19 +18,19 @@ public class TestArticle
     }
 
     public static Article articleFromJson() throws IOException {
-		JsonFactory factory = new JsonFactory();
-    	String s = "{\"content\" : { \"Text\" : \"Peace and love\" } }";
-		JsonParser  p  = factory.createParser(s);
-		p.nextToken();
-		ArticleModelJsonDeserializer ad = new ArticleModelJsonDeserializer();
-    	return ad.deserialize(p, (Article)null);
+        JsonFactory factory = new JsonFactory();
+        String s = "{\"content\" : { \"Text\" : \"Peace and love\" } }";
+        JsonParser p = factory.createParser(s);
+        p.nextToken();
+        ArticleModelJsonDeserializer ad = new ArticleModelJsonDeserializer();
+        return ad.deserialize(p, (Article)null);
     }
 
     public static Article buildArticle() {
-    	Content content = new Content();
-    	content.setText("Peace and love");
-    	Article a = new Article();
-    	a.setContent(content);
-    	return a;
+        Content content = new Content();
+        content.setText("Peace and love");
+        Article a = new Article();
+        a.setContent(content);
+        return a;
     }
 }

--- a/test/java/src/test/java/TestArticle.java
+++ b/test/java/src/test/java/TestArticle.java
@@ -1,0 +1,36 @@
+import static org.junit.Assert.*;
+import java.io.IOException;
+import org.junit.Test;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+
+import io.sysl.issue66.Article;
+import io.sysl.issue66.ArticleModelJsonDeserializer;
+import io.sysl.issue66.Content;
+
+public class TestArticle
+{
+    @Test
+    public void testJson() throws IOException {
+        Article a1 = buildArticle();
+        Article a2 = articleFromJson();
+        assertEquals(a1, a2);
+    }
+
+    public static Article articleFromJson() throws IOException {
+		JsonFactory factory = new JsonFactory();
+    	String s = "{\"content\" : { \"Text\" : \"Peace and love\" } }";
+		JsonParser  p  = factory.createParser(s);
+		p.nextToken();
+		ArticleModelJsonDeserializer ad = new ArticleModelJsonDeserializer();
+    	return ad.deserialize(p, (Article)null);
+    }
+
+    public static Article buildArticle() {
+    	Content content = new Content();
+    	content.setText("Peace and love");
+    	Article a = new Article();
+    	a.setContent(content);
+    	return a;
+    }
+}

